lib/client/signalflow/request_manager.js

```diff
@@ -74,14 +74,18 @@ function RequestManager(endPoint) {
   }
 
   function getJobObject(params) {
+    requiredParams.forEach(function (reqParam){
+      if(!params[reqParam]) {
+        throw new Error('Unable to create a program without required property:'+reqParam);
+      }
+    });
     var obj = {};
-
-    passThroughParams.forEach(function (param) {
-      if (typeof params[param] !== 'undefined') {
-        obj[param] = params[param];
+    Object.keys(params).forEach(function (paramName) {
+      // TODO : we should deep copy here to prevent weirdness if we eventually allow non string/number/boolean values...
+      if(params.hasOwnProperty[paramName]) {
```

Comment:
I don't think you need this check, as `Object.keys` already only returns the intrinsic, iterable properties of the object.